app/src/app/app.component.ts

```diff
@@ -12,11 +12,13 @@ import { EnplugService } from './enplug.service';
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent implements OnInit {
-  constructor(private enplug: EnplugService) {
-  }
-
+  constructor(private enplug: EnplugService) { }
 
   ngOnInit() {
     this.enplug.appStatus.start();
+
+    this.enplug.on('destroy', function (done) {
```

Comment:
[optional] It would be nice to test this call as well. In a test, prepare a mock of `enplug.on` that would store the callback. Run the callback, then check whether the callback's argument (the `done` function) has been called.
Not calling `done` has created problems for us in the past and having a test for that by the default would help immensely.

Reply:
Added test for that